src/certificate/commands/DefectsCommand.ts

```diff
@@ -0,0 +1,150 @@
+import { Service } from 'typedi';
+import { DefectRepository } from '../../defect/DefectRepository';
+import { DefectService } from '../../defect/DefectService';
+import { ITestResult } from '../../models';
+import { ICertificatePayload } from '../../models';
+import { ITestType } from '../../models';
+import { CERTIFICATE_DATA, TEST_RESULTS } from '../../models/Enums';
+import { IDefectParent } from '../../models/IDefectParent';
+import { IFlatDefect } from '../../models/IFlatDefect';
+import { BasePayloadCommand } from '../ICertificatePayloadCommand';
+
+@Service()
+export class DefectsCommand extends BasePayloadCommand {
+	constructor(
+		private defectService: DefectService,
+		private defectRepository: DefectRepository
+	) {
+		super();
+	}
+
+	private certificateIsAnPassOrFail = (): boolean =>
+		this.state.type === CERTIFICATE_DATA.PASS_DATA || this.state.type === CERTIFICATE_DATA.FAIL_DATA;
+
+	public async generate(): Promise<ICertificatePayload> {
+		if (!this.certificateIsAnPassOrFail()) {
+			return {} as ICertificatePayload;
+		}
+
+		const {
+			testResult,
+			testResult: { testTypes },
+		} = this.state;
+
+		const result = {} as ICertificatePayload;
+
+		if (testTypes.testResult !== TEST_RESULTS.FAIL) {
+			result.DATA = {
+				...(await this.getPayloadData(testResult, CERTIFICATE_DATA.PASS_DATA)),
+			};
+		}
+
+		if (testTypes.testResult !== TEST_RESULTS.PASS) {
+			result.FAIL_DATA = {
+				...(await this.getPayloadData(testResult, CERTIFICATE_DATA.FAIL_DATA)),
+			};
+		}
+
+		return result;
+	}
+
+	private async getPayloadData(testResult: ITestResult, type: CERTIFICATE_DATA): Promise<any> {
+		const { isWelsh } = this.state;
+
+		let defectListFromApi: IDefectParent[] = [];
```

Comment:
It looks like `defectListFromApi` is only hydrated inside the if statement. Therefore, if we convert it to a const within the if statement, we save creating the variable for non welsh tests and remove the `let` e.g.

```
let flattenedDefects: IFlatDefect[] = [];

if (isWelsh) {
  const defectListFromApi = await this.defectRepository.getDefectTranslations();
  flattenedDefects = this.defectService.flattenDefectsFromApi(defectListFromApi);
}

// and so on
```